src/main/java/htsjdk/samtools/util/OverlapDetector.java

```diff
@@ -82,13 +82,19 @@ public void addLhs(final T object, final Locatable interval) {
         final int start = interval.getStart() + this.lhsBuffer;
         final int end   = interval.getEnd()   - this.lhsBuffer;
 
-        final Set<T> objects = new HashSet<>(1);
-        objects.add(object);
+        final Set<T> newValue = Collections.singleton(object);
         if (start <= end) {  // Don't put in sequences that have no overlappable bases
-            final Set<T> alreadyThere = tree.put(start, end, objects);
+            final Set<T> alreadyThere = tree.put(start, end, newValue);
             if (alreadyThere != null) {
-                alreadyThere.add(object);
-                tree.put(start, end, alreadyThere);
+                if( alreadyThere.size() == 1){
+                    Set<T> mutableSet = new HashSet<>(2);
```

Comment:
hmmm. better find out if you got a mutable one and just add the new element if so...no?

Reply:
You mean with an instanceof check?  As it's written it should be immutable always if it's exactly size 1 and mutable otherwise.  It's a bit awkward but I thought maybe relying on the size was better than relying on the class.  I can change that though.